File: js/samples/02.messageExtensions.a.searchCommand/src/cards/npmSearchResultCard.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { CardAction, CardFactory, MessagingExtensionAttachment } from 'botbuilder';

/**
 * Creates a messaging extension attachment for an npm search result.
 * @param {any} result The search result to create the attachment for.
 * @returns {MessagingExtensionAttachment} The messaging extension attachment for the search result.
 */
export function createNpmSearchResultCard(result: any): MessagingExtensionAttachment {
    // TODO: Use "Adaptive Card Hero Card" since Hero card is considered deprecated
    const card = CardFactory.heroCard(result.name, [], [], {
        text: result.description
    }) as MessagingExtensionAttachment;
    card.preview = CardFactory.heroCard(result.name, [], [], {
        tap: { type: 'invoke', value: result } as CardAction,
        text: result.description
    });
    return card;
}

File: python/samples/02.messageExtensions.a.searchCommand/src/cards/npm_search_result_card.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from typing import Any, cast

from botbuilder.core import CardFactory
from botbuilder.schema import CardAction, HeroCard
from botbuilder.schema.teams import MessagingExtensionAttachment


def create_npm_search_result_card(result: Any) -> MessagingExtensionAttachment:
    """
    Creates a messaging extension attachment for an npm search result.

    Args:
        result (Any): The search result to create the attachment for.

    Returns:
        MessagingExtensionAttachment: The messaging extension attachment for the search result.
    """
    card = cast(
        MessagingExtensionAttachment,
        CardFactory.hero_card(
            HeroCard(
                title=result["name"],
                images=[],
                buttons=[],
                tap=CardAction(type="invoke", value=result),
                text=result["description"] if "description" in result else "",
            )
        ),
    )
    return card
